src/Permissions/Widgets/PermissionSettingsWidget.vala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2011-2020 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street - Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ * 
+ * Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+ */
+
+public class Permissions.Widgets.PermissionSettingsWidget : Gtk.Box {
+    public Backend.PermissionSettings settings { get; construct set; }
+    public signal void changed_permission_settings (Backend.PermissionSettings settings);
+    public bool do_notify { get; set; default = true; }
+
+    public PermissionSettingsWidget (Backend.PermissionSettings settings) {
+        GLib.Object (
+            settings: settings
+        );
+
+        orientation = Gtk.Orientation.HORIZONTAL;
+        hexpand = true;
+        var vbox = new Gtk.Box (Gtk.Orientation.VERTICAL, 0);
```

Comment:
Instead of nesting boxes, it's usually better to use Gtk.Grid: https://valadoc.org/gtk+-3.0/Gtk.Grid.html